server/neptune/gateway/server.go

```diff
@@ -263,6 +268,36 @@ func NewServer(config Config) (*Server, error) {
 		return nil, errors.Wrap(err, "initializing temporal client")
 	}
 
+	tmpWorkingDir := &source.TmpFileWorkspace{
+		DataDir: config.DataDir,
+	}
+	ghWorkingDir := &source.GithubAppWorkingDir{
+		TmpWorkingDir:  tmpWorkingDir,
+		Credentials:    githubCredentials,
+		GithubHostname: config.GithubHostname,
+		Logger:         ctxLogger,
+	}
+	hookExecutor := &preworkflow.PreWorkflowHookExecutor{}
+	preWorkflowHook := &preworkflow.PreWorkflowHooksRunner{
+		WorkingDir:   ghWorkingDir,
+		GlobalCfg:    globalCfg,
+		HookExecutor: hookExecutor,
+	}
+	//TODO: add metrics
+	clientCreator, err := githubapp.NewDefaultCachingClientCreator(
+		config.App,
+		githubapp.WithClientMiddleware())
```

Comment:
can we move [middleware](https://github.com/lyft/atlantis/blob/release-v0.17.3-lyft.1/server/neptune/workflows/internal/activities/github/middleware.go) out of the internal pkg?